libs/posthog_client_lite/src/lib.rs

```diff
@@ -0,0 +1,433 @@
+//! A lite version of the PostHog client that only supports local evaluation of feature flags.
+
+use std::{collections::HashMap, sync::Arc};
+
+use serde::{Deserialize, Serialize};
+use serde_json::json;
+use sha2::Digest;
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationResponse {
+    #[allow(dead_code)]
+    flags: Vec<PostHogLocalEvaluationFlag>,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlag {
+    key: String,
+    filters: PostHogLocalEvaluationFlagFilters,
+    active: bool,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilters {
+    groups: Vec<PostHogLocalEvaluationFlagFilterGroup>,
+    multivariate: PostHogLocalEvaluationFlagMultivariate,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilterGroup {
+    variant: Option<String>,
+    properties: Option<Vec<PostHogLocalEvaluationFlagFilterProperty>>,
+    rollout_percentage: f64,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagFilterProperty {
+    key: String,
+    value: PostHogFlagFilterPropertyValue,
+    operator: String,
+}
+
+#[derive(Serialize, Deserialize)]
+#[serde(untagged)]
+pub enum PostHogFlagFilterPropertyValue {
+    String(String),
+    Number(f64),
+    Boolean(bool),
+    List(Vec<String>),
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariate {
+    variants: Vec<PostHogLocalEvaluationFlagMultivariateVariant>,
+}
+
+#[derive(Deserialize)]
+pub struct PostHogLocalEvaluationFlagMultivariateVariant {
+    key: String,
+    rollout_percentage: f64,
+}
+
+pub struct PostHogClient {
+    server_api_key: String,
```

Comment:
It's not clear what is public/private api, what is server/client API key. Probably it's described in the docs somewhere, but by just reading the code it's not clear. Let's add some description or put a link to the docs nearby

Reply:
added a commit to doc this